src/app-layer-parser.c

```diff
@@ -140,6 +140,8 @@ struct AppLayerParserState_ {
      * we don't need a var per direction since we don't log a transaction
      * unless we have the entire transaction. */
     uint64_t log_id;
+    /* Count of number of transactions */
+    uint64_t tx_cnt;
```

Comment:
Why does this need to be part of the state? It's only used in AppLayerParserParse so it can be a local variable.